pkg/operator/cluster/controller.go

```diff
@@ -158,16 +179,18 @@ func (c *ClusterController) onAdd(obj interface{}) {
 		return
 	}
 
+	c.initializedClusters[clust.Namespace] = cluster
```

Comment:
help me understand the book keeping of `initializedClusters` better.  if the operator pod restarts, does it repopulate this list?  if not, do we leak anything (i.e. do we lost the ability to properly update or clean up a cluster later on)?

Reply:
yes, when the operator restarts, it initializes each CRD again to 1) make sure the latest desired state is applied and 2) to watch for file/object/pool CRDs for each cluster. So the `initializedClusters` will keep track of that initialization once per cluster since the operator started. If the operator is restarted, it will add it back to the initialized list again.